### PR TITLE
[WIP] Roadmap update

### PR DESCRIPTION
**Pull request by rebornix**
Yay! We love PRs! 🎊

Please include a description of your change & check your PR against this list, thanks:

- [x] Commit message has a short title & issue references
- [x] Each commit does a logical chunk of work.  
- [x] It builds and tests pass (e.g `gulp tslint`)

I'm going through the roadmap and see our current status. But I do find the warning tag a little bit confusing. It indicates the feature is partly implemented but we don't which part is missing: corner user case, test case or anything else? When we run into issues with commands in this category, we don't know whether it's a bug or a missing feature.

So here I did two things to the roadmap

1. Add 🔴 to each command that requires better description of the status: which part is missing and how current command works. Part of the commands are handled by me so I'll give them a update while rest of them are implemented by @jpoon @johnfn @xconverge @sectioneight @aminroosta ,etc. Please leave comments about the status. I'll add them to the PR and then remove the 🔴 mark.
2. Add ⬇️ to most features which rank low on our list.

After above update, we are supposed to have

1. Most commands covered, whether it's done, partially finished with detailed description, or low-pritory.
2. The rest of them will come soon.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -109,39 +111,39 @@ Status | Command | Description
 
 Status | Command | Description
 ---|--------|------------------------------
-:warning: | :1234: `/{pattern}[/[offset]]<CR>` | search forward for the Nth occurrence of {pattern}
-:warning: | :1234: `?{pattern}[?[offset]]<CR>` | search backward for the Nth occurrence of {pattern}
+:warning: :red_circle: | :1234: `/{pattern}[/[offset]]<CR>` | search forward for the Nth occurrence of {pattern}
+:warning: :red_circle: | :1234: `?{pattern}[?[offset]]<CR>` | search backward for the Nth occurrence of {pattern}
  | :1234: `/<CR>` | repeat last search, in the forward direction
```
Comment (xconverge) [inline ROADMAP.md]: green checkmark, this is implemented

File: ROADMAP.md [hunk 2]
```diff
@@ -109,39 +111,39 @@ Status | Command | Description
...
  | :1234: `?<CR>` | repeat last search, in the backward direction
```
Comment (xconverge) [inline ROADMAP.md]: green checkmark, this is implemented

File: ROADMAP.md [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
 :1234: - command accepts numeric prefix
 
+:red_circle: - command accessment needs review
```
Comment (sectioneight) [inline ROADMAP.md]: typo: assessment

**Comment by sectioneight** [comment]
I've looked over all the red circles and verified that I don't have updates. I believe I've kept the roadmap up to date with all my PRs. Will take another pass at my PRs tomorrow and verify there isn't something this PR is missing.

**Comment by rebornix** [comment]
@sectioneight thanks for your quick update :)

File: ROADMAP.md [hunk 4]
```diff
@@ -217,8 +219,8 @@ in Visual block mode:
 
 Status | Command | Description
 ---|--------|------------------------------
-   | I	| insert the same text in front of all the selected lines
-   | A	| append the same text after all the selected lines
+:red_circle:| I	| insert the same text in front of all the selected lines
```
Comment (xconverge) [inline ROADMAP.md]: implemented in #720

File: ROADMAP.md [hunk 5]
```diff
@@ -217,8 +219,8 @@ in Visual block mode:
...
+:red_circle:| A	| append the same text after all the selected lines
```
Comment (xconverge) [inline ROADMAP.md]: implemented/fixed in #720

File: ROADMAP.md [hunk 6]
```diff
@@ -107,41 +109,41 @@ Status | Command | Description
...
-:warning: | :1234: `?{pattern}[?[offset]]<CR>` | search backward for the Nth occurrence of {pattern}
- | :1234: `/<CR>` | repeat last search, in the forward direction
- | :1234: `?<CR>` | repeat last search, in the backward direction
-:warning: | :1234: n | repeat last search
-:warning: | :1234: N | repeat last search, in opposite direction
+Status | Command | Description | Note
+---|--------|------------|------------------
+:warning: | :1234: `/{pattern}[/[offset]]<CR>` | search forward for the Nth occurrence of {pattern} | Currently we only support JavaScript Regex but not Vim's in-house Regex engine.
+:warning: | :1234: `?{pattern}[?[offset]]<CR>` | search backward for the Nth occurrence of {pattern} | Currently we only support JavaScript Regex but not Vim's in-house Regex engine.
+:white_check_mark: | `/<CR>` | repeat last search, in the forward direction
```
Comment (xconverge) [inline ROADMAP.md]: sorry, this is actually half implemented, it is supposed to have the num option that you removed.

Both this and the following are implemented for the non num preceded version so maybe yellow hazard icon?

File: ROADMAP.md [hunk 7]
```diff
@@ -107,41 +109,41 @@ Status | Command | Description
...
+:white_check_mark: | :1234: `?<CR>` | repeat last search, in the backward direction
```
Comment (xconverge) [inline ROADMAP.md]: see note above

File: ROADMAP.md [hunk 8]
```diff
@@ -377,12 +379,11 @@ Status | Command | Description
...
 :white_check_mark:   | V		| start highlighting linewise
-   | CTRL-V	| start highlighting blockwise
-   | o		| exchange cursor position with start of highlighting
-   | gv		| start highlighting on previous visual area
+:white_check_mark:| o		| exchange cursor position with start of highlighting
+:arrow_down: | gv		| start highlighting on previous visual area
 :white_check_mark:   | v		| highlight characters or stop highlighting
 :white_check_mark:   | V		| highlight linewise or stop highlighting
-   | CTRL-V	| highlight blockwise or stop highlighting
+:red_circle: | CTRL-V	| highlight blockwise or stop highlighting
```
Comment (xconverge) [inline ROADMAP.md]: should be good

File: ROADMAP.md (hunk 6 above)
Comment (rebornix) [inline ROADMAP.md]: Sure, so the status is `{count} is not supported`.

**Comment by rebornix** [comment]
@johnfn I noticed you converted the state of `y{motion}`, `c{motion]`, etc from ✅ to ⚠️  , can you help me understand what's the catch there?

File: ROADMAP.md [hunk 9]
```diff
@@ -298,14 +300,14 @@ Status | Command | Description
 
 Status | Command | Description
 ---|--------|------------------------------
-:warning:   | "{char}	        | use register {char} for the next delete, yank, or put
+:warning: :red_circle:  | "{char}	        | use register {char} for the next delete, yank, or put
```
Comment (rebornix) [inline ROADMAP.md]: @aminroosta you implement these features and I'm seeing them working well. Is there any hidden issue with it which makes you mark it as ⚠️ ?

Comment (aminroosta) [inline ROADMAP.md]: @rebornix I think this is compele. But there are other low priority read only registers like command line register that are not implemented.

I've left some comments in register.ts file, I will send PR updating roadmap and adding them as low priority.

Comment (rebornix) [inline ROADMAP.md]: Thanks for your info, I'll mark it as `read only registers not supported yet` first :) Then you may want to give it a better description if you like.

**Comment by johnfn** [comment]
@rebornix I think that might have been a mistake and I probably intended to mark yy and cc with ⚠️ . In any case they appear to be working now. 

I may have marked y with ⚠️  because I was concerned about undo/redo, but I'm more confident about that now, too.

**Comment by rebornix** [comment]
@johnfn thanks for the input! I just added some new icons to the roadmap and the whole idea is making the status of commands more positive and instructive. A warning is somewhat confusing as people have no idea about the expected behavior of the commands.

I added below two signs

* :white_check_mark: :star: telling users this command is done and it's somewhat customized with Code and we think it's better this way instead of making it totally the same as Vim.
* :running: It's in progress, whether there is already a PR/issue for it or it's on our task board (mine, yours, etc).

Note 🔴 will be removed once we finish the assessment of this roadmap. I'll say we are really close in some extent (comparing to competitors on other platforms) and my next step is looking into commands which are not implemented yet and see if they need more Code's ❤️ .

**Pull request merged**